felix/bpf-gpl/tc.c

```diff
@@ -937,20 +938,26 @@ static CALI_BPF_INLINE enum do_nat_res do_nat(struct cali_tc_ctx *ctx,
 	return NAT_ALLOW;
 
 icmp_too_big:
-	state->icmp_type = ICMP_DEST_UNREACH;
-	state->icmp_code = ICMP_FRAG_NEEDED;
+#ifndef IPVER6
+	STATE->icmp_type = ICMP_DEST_UNREACH;
+	STATE->icmp_code = ICMP_FRAG_NEEDED;
 
 	struct {
 		__be16  unused;
 		__be16  mtu;
 	} frag = {
 		.mtu = bpf_htons(TUNNEL_MTU),
 	};
-	state->tun_ip = *(__be32 *)&frag;
+	STATE->tun_ip = *(__be32 *)&frag;
 
 	return NAT_ICMP_TOO_BIG;
+#else
+	/* XXX */
```

Comment:
Can you expand these XXX todos a bit so it's easier to know what's missing in future.  Easy for that to get lost without a signpost!